WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -60,15 +77,35 @@ public void onCreate(Bundle savedInstanceState) {
 
         setupActionBar();
         setupContent();
+        initViewModel();
         handleIntent(getIntent());
     }
 
     private void setupContent() {
         mPager = findViewById(R.id.postPager);
-        mPostsPagerAdapter = new PostsPagerAdapter(mSite, this, getSupportFragmentManager());
+        mPostsPagerAdapter = new PostsPagerAdapter(PostListMainViewModelKt.getPOST_LIST_PAGES(), mSite, this,
+                getSupportFragmentManager());
         mPager.setAdapter(mPostsPagerAdapter);
         TabLayout tabLayout = findViewById(R.id.tabLayout);
         tabLayout.setupWithViewPager(mPager);
+
+        mPager.addOnPageChangeListener(new OnPageChangeListener() {
```

Comment:
Since this adds a new listener, should we clear the listeners before adding the new listener? I know `setupContent` is called only once right now, so it won't make a difference, but it wouldn't hurt to be defensive about a potential change. What do you think?

Reply:
Tbh I'm not a fan of this kind of defensive approach. Whenever I read a code and I encounter something like this and I can't think of a reason why the code is there, it just confuses me. I'm afraid of touching the code, because I feel like I don't understand it well enough. However, I see your point and I'm ok to add it if we also add a comment with an explanation (any suggestions are welcome:D).

Reply:
I think we can just say `To avoid having multiple listeners, let's clear the existing ones if there are any` or something similar. Having said that, it's totally your call for whether to add it or not. It just makes me a bit uneasy that we might be adding multiple listeners and I feel the need to look at the rest of the code to ensure this method will be called only once during the lifecycle of the activity.

Reply:
Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/9255